Triangulation_2/doc/Triangulation_2/Concepts/TriangulationTraits_2.h

```diff
@@ -187,29 +187,6 @@ typedef unspecified_type Construct_circumcenter_2;
 
 /// @}
 
-/// \name Creation
-/// Only a default constructor, copy constructor and an assignment
-/// operator are required. Note that further constructors can be
-/// provided.
-/// @{
-
-/*!
-default constructor.
-*/
-TriangulationTraits_2();
-
-/*!
-Copy constructor
-*/
-TriangulationTraits_2(TriangulationTraits_2 gtr);
-
-/*!
-Assignment operator.
-*/
-TriangulationTraits_2 operator=(TriangulationTraits_2 gtr);
-
-/// @}
-
```

Comment:
If your new model doesn't fit requirements, the solution is not to strip the concept from the requirements your model doesn't honor... These are real requirements that are not to be removed.

Reply:
No, when testing the new concept, we realized with @sloriot that all those requirements were unnecessary, and he asked me to remove them.

Reply:
Pretty sure that (for example):
```
  Triangulation_2(const Geom_traits& geom_traits=Geom_traits());
```
(https://github.com/CGAL/cgal/blob/master/Triangulation_2/include/CGAL/Triangulation_2.h#L244) requires the geometric traits to have a default constructor. But you are removing that from the concept.

Reply:
Default constructor is not needed, even if the triangulation have that default value)

Reply:
How is it not needed when you have `Geom_traits()` literally written in the constructor? Just try to compile `T2 tr;` with this projection traits and you will see that it does not compile...

Reply:
I think it is only required for user using the default value. And thus that is not strictly required.

Reply:
Exaggerating, `Side_of_oriented_circle_2` is also only required if I insert points so it is not strictly required either.

Concepts should describe the requirements to use the complete code, not some arbitrary subset based on some random user code.

Reply:
I agree in general but not for the default constructor. I prefer having a compilation error than having some runtime bugs because I forgot to pass the constructed traits. If the user uses a traits that is not default constructible then she shall not use the triangulation default constructor.

Reply:
Again, your model being only a partial fit of the concept is not a valid reason to strip the concept. And it strips of more than the default constructor.

You can leave your model not being compatible with some triangulation functionalities if you wish (although I think it would be cleaner to add whatever is missing and explicitly disable and document), just do not modify the concept.

Reply:
Concepts should be kept minimal. If, as I hope, CGAL manages to implement C++20 concepts, then concepts issues will become compilation errors.

Taking an example from the STL, in C++03, the requirements for `T` in `std::vector<T>` was:
> `T` must meet the requirements of `CopyAssignable` and `CopyConstructible`.

... but `std::vector<T> vec(20);` would also require that `T()` is valid, to use the default-value of `value` in that constructor:
```c++
explicit vector( size_type count,
                 const T& value = T(),
                 const Allocator& alloc = Allocator());
```

<details>
<summary>Why C++03?</summary>

I specify C++03, because in C++11 or later, the requirement have become
> The requirements that are imposed on the elements depend on the actual operations performed on the container. Generally, it is required that element type meets the requirements of `Erasable`, but many member functions impose stricter requirements. 

</details>

Reply:
The traits class has been here for years and worked with the triangulation. The concept is just too restrictive requesting things that are not needed.

Reply:
A complete code of
```
struct Gt { };
int main() { Triangulation_2<Gt> tr; }
```
can also be put in CGAL and work for years, will that prove that all the concept `TriangulationTraits_2` is requesting things that are not needed? Partial traits can work with a partial usage of the triangulation, just like your partial model of projection traits have worked because it only uses a part of the triangulation.

The concept is not too restrictive, it covers requirements of all the functionalities. Feel free to compile all of Triangulation_2 with your projection traits (starting as I have already said in https://github.com/CGAL/cgal/pull/5407#discussion_r639477557 with the default constructor) and you will see why these requirements exist.

Reply:
I don't understand what's your problem here. Sure it will not compile as the traits has no default constructor. But that the same thing with `std::vector<T>::resize(n)`. It will not compile if `T` is not default constructible.

Reply:
Agreement: Do not modify the concept and kept the doc of the projection traits as is, even if it is not entirely true.